Frameworks/CoreImage/CIImage.mm

```diff
@@ -73,90 +74,164 @@ @implementation CIImage
 /**
  @Status Interoperable
 */
-- (CIImage*)init {
++ (CIImage*)emptyImage {
+    return [[[CIImage alloc] init] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (CGRect)extent {
+    return self->_extent;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
-        _cgImage = nil;
-        _color = nil;
-        _filter = nil;
         _extent = CGRectInfinite;
     }
     return self;
 }
 
+- (instancetype)_initWithCGImage:(CGImageRef)image color:(CIColor*)color {
```

Comment:
Since you are introducing this private initializer, you should absolutely:

1. Make sure the public initializers use THIS initializer.
2. Do not use public initializers from this initializer.

This should be the end of the line.